lib/iris/analysis/trajectory.py

```diff
@@ -272,112 +279,115 @@ def interpolate(cube, sample_points, method=None):
-            # Check structure of the indexed coord (at one selected point).
-            point_coord = single_point_test_cube.coord(src_coord)
-            if len(point_coord.points) != 1:
-                msg = ('Coord {} at one x-y position has the shape {}, '
-                       'instead of being a single point. ')
-                raise ValueError(msg.format(src_coord.name(), src_coord.shape))
-
-            # Work out which indices apply to the input coord.
-            # NOTE: we know how to index the source cube to get a cube with a
-            # single point for each coord, but this is very inefficient.
-            # So here, we translate cube indexes into *coord* indexes.
-            src_coord_dims = cube.coord_dims(src_coord)
-            fancy_coord_index_arrays = [list(all_point_indices[:, src_dim])
-                                        for src_dim in src_coord_dims]
-
-            # Fill the new coord with all the correct points from the old one.
-            new_cube_coord.points = src_coord.points[fancy_coord_index_arrays]
-            # NOTE: the new coords do *not* have bounds.
-
-    return new_cube
+            # Fill in the empty squashed (non derived) coords.
+            column_coords = [coord
+                             for coord in cube.dim_coords + cube.aux_coords
+                             if not squish_my_dims.isdisjoint(
+                                 cube.coord_dims(coord))]
+            new_cube_coords = [new_cube.coord(column_coord.name())
+                               for column_coord in column_coords]
+            all_point_indices = np.array(column_indexes)
+            single_point_test_cube = cube[column_indexes[0]]
+            for new_cube_coord, src_coord in zip(new_cube_coords, column_coords):
+                # Check structure of the indexed coord (at one selected point).
+                point_coord = single_point_test_cube.coord(src_coord)
+                if len(point_coord.points) != 1:
+                    msg = ('Coord {} at one x-y position has the shape {}, '
+                           'instead of being a single point. ')
+                    raise ValueError(msg.format(src_coord.name(), src_coord.shape))
+
+                # Work out which indices apply to the input coord.
+                # NOTE: we know how to index the source cube to get a cube with a
+                # single point for each coord, but this is very inefficient.
+                # So here, we translate cube indexes into *coord* indexes.
+                src_coord_dims = cube.coord_dims(src_coord)
+                fancy_coord_index_arrays = [list(all_point_indices[:, src_dim])
+                                            for src_dim in src_coord_dims]
+
+                # Fill the new coord with all the correct points from the old one.
+                new_cube_coord.points = src_coord.points[fancy_coord_index_arrays]
+                # NOTE: the new coords do *not* have bounds.
+
+    if not index_only:
```

Comment:
Also not necessary as the code should not get this far if the index_only method is True.